lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -240,13 +255,36 @@ class ForLoopLoweringStateInterface
   }
 };
 
+class PipeOpLoweringStateInterface {
```

Comment:
I'm not very satisfied with this approach though, having to introduce another interface to "remember" the register set for the sequential/pipelined library operator.

Reply:
Well, we'll definitely need some way to recall this mapping, since it isn't explicitly stated in the IR. If this is only necessary for lowering conditionals in `if` statements, then I think it would make more sense in the `IfLoweringStateInterface`.

With that said, I don't think this is the only Calyx control flow that will require this special treatment, i.e., `while` should be similar. What other approaches are you thinking about?

In any case, it would be nice to add documentation on why this interface exists.